sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/FromUnsafe.scala

```diff
@@ -52,6 +53,8 @@ case class FromUnsafe(child: Expression) extends UnaryExpression
       }
       new GenericArrayData(result)
 
+    case StringType => value.asInstanceOf[UTF8String].clone()
```

Comment:
For top level `UTF8String` in `UnsafeRow` field, yes, won't execute.
But for inner `UTF8String` inside array/map/struct, will execute, as we call this `convert` method recursively.